qiskit/visualization/circuit_visualization.py

```diff
@@ -335,6 +335,12 @@ def _text_circuit_drawer(
     text_drawing.vertical_compression = vertical_compression
 
     if filename:
+        if not filename.endswith(".txt"):
+            logger.warning(
```

Comment:
This should raise an error not log a warning. so something like `raise ValueError("error message")`